SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/ClientServerRPCService.cpp

```diff
@@ -0,0 +1,376 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/RPCs/ClientServerRPCService.h"
+
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/ClientEndpoint.h"
+#include "Schema/ServerEndpoint.h"
+#include "SpatialConstants.h"
+#include "Utils/RepLayoutUtils.h"
+
+DEFINE_LOG_CATEGORY(LogClientServerRPCService);
+
+namespace SpatialGDK
+{
+ClientServerRPCService::ClientServerRPCService(const ExtractRPCDelegate InExtractRPCCallback, const FSubView& InSubView,
+											   USpatialNetDriver* InNetDriver, FRPCStore& InRPCStore)
+	: ExtractRPCCallback(InExtractRPCCallback)
+	, SubView(&InSubView)
+	, NetDriver(InNetDriver)
+	, RPCStore(&InRPCStore)
+{
+}
+
+void ClientServerRPCService::Advance()
+{
+	const FSubViewDelta& SubViewDelta = SubView->GetViewDelta();
+	for (const EntityDelta& Delta : SubViewDelta.EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::UPDATE:
+		{
+			for (const ComponentChange& Change : Delta.ComponentUpdates)
+			{
+				ComponentUpdate(Delta.EntityId, Change.ComponentId, Change.Update);
+			}
+			break;
+		}
+		case EntityDelta::ADD:
+			PopulateDataStore(Delta.EntityId);
+			EntityAdded(Delta.EntityId);
+			break;
+		case EntityDelta::REMOVE:
+			ClientServerDataStore.Remove(Delta.EntityId);
+			break;
+		case EntityDelta::TEMPORARILY_REMOVED:
```

Comment:
Don't you need to try to extract RPCs as well in that case ?

Reply:
yep! Totally do. Thanks for pointing it out